apps/api/src/applicant/external-api.service.ts

```diff
@@ -105,4 +108,101 @@ export class ExternalAPIService {
       this.logger.log(`Error in saveReasons(): ${e}`);
     }
   }
+
+  /**
+   * fetch and upsert applicant details
+   */
+  async saveApplicant(): Promise<void> {
+    try {
+      const data = await this.external_request.getApplicants();
+      await this.createBulkApplicants(data);
+    } catch (e) {
+      this.logger.error({ e });
+    }
+  }
+
+  /**
+   * fet user/staff and HA details to reduce database calls.
+   * @returns
+   */
+  async getApplicantMasterData() {
+    // fetch user/staff details
+    const usersArray = await this.ienUsersRepository.find();
+    const users: any = {};
+    usersArray.map(user => {
```

Comment:
map return an array but it isn't used. how about `forEach` instead of `map`?

Reply:
forEach is correct here, Just fix it and pushed new code.